src/nerdbank-streams/src/tests/MultiplexingStream.SeededChannels.spec.ts

```diff
@@ -79,9 +79,13 @@ async function readAsync(readable: NodeJS.ReadableStream): Promise<Buffer | null
     if (readBuffer === null) {
         const bytesAvailable = new Deferred<void>();
         const streamEnded = new Deferred<void>();
-        readable.once("readable", bytesAvailable.resolve.bind(bytesAvailable));
-        readable.once("end", streamEnded.resolve.bind(streamEnded));
+        const bytesAvailableCallback = bytesAvailable.resolve.bind(bytesAvailable);
+        const streamEndedCallback = streamEnded.resolve.bind(streamEnded);
+        readable.once("readable", bytesAvailableCallback);
+        readable.once("end", streamEndedCallback);
         await Promise.race([bytesAvailable.promise, streamEnded.promise]);
+        readable.removeListener("readable", bytesAvailableCallback);
+        readable.removeListener("end", streamEndedCallback);
```

Comment:
Can you help me understand the significance of these test changes? It seems that both files go to greater lengths to remove listeners, but how does that improve the test? Is it relevant to exposing the bug that you're fixing?

Reply:
no, not really, when we have two listeners and only wait on one of them, the other one will be leaked. there are quite some memory leak warning messages during running unit tests and trying it in product environment. when tests ran out of memory, I tried to get rid of memory leaks based on those messages first. The unchain event changes did eliminate those memory leaking warnings, so I leave them here although it was not the reason of the failure in the first place.  It might not be necessary for all the testing code, but those leaking messages doesn't tell the exact location, so preventing them to happen could help us to capture new issues.  

I guess you might not have seen those leaking warnings. It could be I am using the new version of NodeJS. It prints out warnings when extra number of event handlers were added to a same source, which indicates a potential leaking.

I guess it is annoying pattern to repeat, so maybe we should consider some infrastructure to keep it simple. But, just get into typeScript world, so I guess it is better to keep it simple.